formats/json/jsTest/src/kotlinx/serialization/json/DynamicPolymorphismTest.kt

```diff
@@ -103,6 +115,129 @@ class DynamicPolymorphismTest {
         }
     }
 
+    @Test
+    fun testNullable() {
+        val nonNullChild = Sealed.NullableChild("nonnull")
+        encodeAndDecode(Sealed.serializer(), nonNullChild, arrayJson) {
+            assertEquals("nullable_child", this[0])
+            val dynamicValue = this[1]
+            assertEquals(nonNullChild.nullable, dynamicValue.nullable)
+            assertEquals(nonNullChild.intField, dynamicValue.intField)
+            assertEquals(2, fieldsCount(dynamicValue))
+        }
+        encodeAndDecode(Sealed.serializer(), nonNullChild, objectJson) {
+            assertEquals("nullable_child", this.type)
+            assertEquals(nonNullChild.nullable, this.nullable)
+            assertEquals(nonNullChild.intField, this.intField)
+            assertEquals(3, fieldsCount(this))
+        }
+
+        val nullChild = Sealed.NullableChild(null)
+        encodeAndDecode(Sealed.serializer(), nullChild, arrayJson) {
+            assertEquals("nullable_child", this[0])
+            val dynamicValue = this[1]
+            assertEquals(nullChild.nullable, dynamicValue.nullable)
+            assertEquals(nullChild.intField, dynamicValue.intField)
+            assertEquals(2, fieldsCount(dynamicValue))
+        }
+        encodeAndDecode(Sealed.serializer(), nullChild, objectJson) {
+            assertEquals("nullable_child", this.type)
+            assertEquals(nullChild.nullable, this.nullable)
+            assertEquals(nullChild.intField, this.intField)
+            assertEquals(3, fieldsCount(this))
+        }
```

Comment:
This test is specifically what was failing before. It was throwing a `TypeError: Cannot convert undefined or null to object` when we tried `js("Object").keys(value)`